### PR TITLE
Advanced transactions max; wallet selection fix

### PR DESCRIPTION
**Pull request by yzhurko**

### CODE REVIEW

File: GUI/QtQuick/qml/Send/AdvancedDetails.qml [hunk 1]
```diff
@@ -440,6 +440,14 @@ ColumnLayout  {
                     width: BSSizes.applyScale(504)
                     height: BSSizes.applyScale(70)
 
+                    function getMax() {
+                        if (!isRBF && !isCPFP && txInputsSelectedModel.rowCount > 1) {
+                            return (txInputsSelectedModel.sourceModel.balance - txOutputsModel.totalAmount).toFixed(8)
```
Comment (sergey-chernikov) [inline GUI/QtQuick/qml/Send/AdvancedDetails.qml]: This is not the correct max amount calculation, you miss the expected fee here. You need to set the correct params for tempRequest retrieval instead.

**Comment by yzhurko** [comment]
found one more issue in these changes

File: GUI/QtQuick/AddressFilterModel.cpp [hunk 2]
```diff
@@ -82,21 +82,18 @@ bool AddressFilterModel::filterAcceptsRow(int source_row, const QModelIndex& sou
...
-      }
+       const auto leftAmount = sourceModel()->data(sourceModel()->index(left.row(), 2)
+           , QmlAddressListModel::TableRoles::TableDataRole).toDouble();
+       const auto leftIndex = sourceModel()->data(sourceModel()->index(left.row(), 0)
+           , QmlAddressListModel::TableRoles::AddressTypeRole).toString().remove(0, 2).toInt();
+       const auto rightAmount = sourceModel()->data(sourceModel()->index(right.row(), 2)
+           , QmlAddressListModel::TableRoles::TableDataRole).toDouble();
+       const auto rightIndex = sourceModel()->data(sourceModel()->index(right.row(), 0)
+           , QmlAddressListModel::TableRoles::AddressTypeRole).toString().remove(0, 2).toInt();
+       if (qFuzzyIsNull(leftAmount) ^ qFuzzyIsNull(rightAmount)) {
```
Comment (sergey-chernikov) [inline GUI/QtQuick/AddressFilterModel.cpp]: What's the meaning of bitwise XOR for boolean operands? Did you mean `!=` instead?

File: GUI/QtQuick/QTXSignRequest.cpp [hunk 3]
```diff
@@ -81,6 +81,9 @@ QStringList QTXSignRequest::outputAddresses() const
    for (const auto& recip : txReq_.getRecipients([](const bs::Address&) { return true; })) {
       try {
          const auto& addr = bs::Address::fromRecipient(recip);
+         if (addr == txReq_.change.address) {
+             continue;
+         }
```
Comment (sergey-chernikov) [inline GUI/QtQuick/QTXSignRequest.cpp]: I think you should apply this to lambda filter 3 lines above (which always returns `true` currently) - it was designed exactly for this.

File: GUI/QtQuick/QTXSignRequest.cpp [hunk 4]
```diff
@@ -100,6 +103,29 @@ QString QTXSignRequest::outputAmount() const
...
+   if (!txReq_.isValid()) {
+      return {};
+   }
+   QStringList result;
+   for (const auto& recip : txReq_.getRecipients([](const bs::Address&) { return true; })) {
+       try {
+           const auto& recipAddr = bs::Address::fromRecipient(recip);
+           if (recipAddr == txReq_.change.address) {
+               continue;
+           }
```
Comment (sergey-chernikov) [inline GUI/QtQuick/QTXSignRequest.cpp]: same lambda filter above

**Review by sergey-chernikov — APPROVED** [review]

**Pull request merged**